plugins/luxtronic2/__init__.py

```diff
@@ -328,22 +399,44 @@ def _decode(self, identifier, value):
         return value
 
     def parse_item(self, item):
+        def __reverseListOp(param):
+            # reversing list building of smarthome.py config parser for the
+            # binary or opperator |
```

Comment:
@ohinckel I just check the config parser again and quotes are not supported right now. I also tried it and there is no magic gunk happening somewhere else.

config.py line:89...
```python
attr, __, value = line.partition('=')
...
attr = attr.strip()
...
if '|' in value:
    item[attr] = [strip_quotes(x) for x in value.split('|')]
else:
    item[attr] = strip_quotes(value)
```

So to allow list operators in combination with quotes, it would have to ignore pipes inside of quotes and split if there are pips outside of them. I did a quick hack, see below. If you have a better solution or if I missed something let me know.

```python
def split_pipe(s):
    output = []
    split_pos = []
    quotes = ['"',"'"]
    pipe = '|'
    in_quoate = False
    start_quote = None
    
    for i in range(0, len(s)):
        if not in_quoate and s[i] == pipe:
            split_pos.append(i)
        
        if not in_quoate and s[i] in quotes:
            start_quote = s[i]
            in_quoate = True  
        elif in_quoate and s[i] == start_quote:
            in_quoate = False
            
    if len(split_pos) > 0:      
        output.append(s[0:split_pos[0]])
        for i in range(0,len(split_pos)):
            if i+1 < len(split_pos):
                output.append(s[split_pos[i]+1:split_pos[i+1]])
            else:
                output.append(s[split_pos[i]+1:])
        output = [strip_quotes(x) for x in output]
        return output
    else:
        return strip_quotes(s)

s = """'123|456'|"789'10'|1112"|131415 """

current = [strip_quotes(x) for x in s.split('|')]
for i in current:
    print(i)
print('-'*10)
x = split_pipe(s)
for i in x:
    print(i)
print('-'*10)
x = split_pipe('1')
print(x)
```